File: rtc4j/src/main/java/com/rapidtable/sdk/rtc4j/resource/project/ProjectResponse.java
/*
 * Copyright Rapid Table, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License. A copy of the License is located at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

package com.rapidtable.sdk.rtc4j.resource.project;

/**
 * Project Response model
 */
public class ProjectResponse {
    private String id;
    private String name;
    private String description;
    private String theme;
    private String coverPath;
    private Boolean favorite;

    public ProjectResponse() {
    }

    public ProjectResponse(String id, String name, String description, String theme, String coverPath, Boolean favorite) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.theme = theme;
        this.coverPath = coverPath;
        this.favorite = favorite;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getTheme() {
        return theme;
    }

    public void setTheme(String theme) {
        this.theme = theme;
    }

    public String getCoverPath() {
        return coverPath;
    }

    public void setCoverPath(String coverPath) {
        this.coverPath = coverPath;
    }

    public Boolean getFavorite() {
        return favorite;
    }

    public void setFavorite(Boolean favorite) {
        this.favorite = favorite;
    }
}

File: rtc4js/src/resource/project/project-response.ts
/*
 * Copyright Rapid Table, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the 'License').
 * You may not use this file except in compliance with the License. A copy of the License is located at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

/**
 * Project Resource Response model
 */
export class ProjectResponse {
  constructor(
    public id: string = '',
    public name: string = '',
    public description: string = '',
    public theme: string = '',
    public coverPath: string = '',
    public favorite: boolean = false
  ) {}

  public static of({
    id,
    name,
    description,
    theme,
    coverPath,
    favorite,
  }: ProjectResponse): ProjectResponse {
    return new ProjectResponse(
      id,
      name,
      description,
      theme,
      coverPath,
      favorite
    );
  }
}
